CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeEventManager.cs

```diff
@@ -285,9 +286,10 @@ internal void NotifyUserProfileInitialized()
                 { "CleverTapID",  _coreState.DeviceInfo.DeviceId },
                 { "CleverTapAccountID", _accountId }
             };
+            _callbackHandler.CleverTapProfileInitializedCallback(Json.Serialize(eventInfo));
```

Comment:
Move the logic for updating the storage id and the initialized callback in `SwitchOrCreateProfile` since the `NotifyUserProfileInitialized` method is called on multiple occasions and is specific for the Profile Initialized callback.